### DISCUSSION TITLE
How to list all installed versions of a versioned formula?

### DISCUSSION THREAD

**Discussion by mislav** [Homebrew/discussions#4845] [Everyday usage] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I have installed multiple different `@`-versions of the `openssl` formula:
```
$ `brew --prefix openssl@1.1`/bin/openssl version
OpenSSL 1.1.1v  1 Aug 2023

$ `brew --prefix openssl@3.0`/bin/openssl version
OpenSSL 3.0.11 19 Sep 2023 (Library: OpenSSL 3.0.11 19 Sep 2023)

$ `brew --prefix openssl@3`/bin/openssl version
OpenSSL 3.1.3 19 Sep 2023 (Library: OpenSSL 3.1.3 19 Sep 2023)
```
For scripting purposes, I'm now trying to find a `brew` command incantation that can **list all the different `openssl@*` formulae that I have installed**.

This does _not_ do what I expect:
```
$ brew list --versions openssl
openssl@3 3.1.2 3.1.3
```
This lists all installed "versions" of the `openssl@3` formula, but these are not the droids I'm looking for. I'm looking for all installed `openssl@*` [versioned formulae](https://docs.brew.sh/Versions) and don't want to scope the search to `openssl@3` specifically.

This gets me close, but seems wasteful/wrong:
```
$ brew list | grep ^openssl@
openssl@1.1
openssl@3
openssl@3.0
```
I imagine that `brew list` could get relatively slow on systems where there a ton of formula installed, but that's only a guess. Ultimately, I don't want to list all installed formula; I just want ones named `openssl@*`. It feels wrong to have to perform filtering with grep rather than having Homebrew do the filtering for me.

Does anyone know of a better way?

**Comment by SMillerDev** [comment, marked as answer]
No, that seems like the best way to me. `brew list` is just `ls` under the hood, so it should be pretty quick.
